README.md

```diff
@@ -80,6 +80,48 @@ IMG_NAME='Raspbian'
 ```
 
 
+## How the build process works
+
+The following process is followed to build images:
+
+ * Loop through all of the stage directories in alphanumeric order
+
+ * Move on to the next directory if this stage directory contains a file called
+   "SKIP"
+
+ * Run the script ```prerun.sh``` which is generally just used to copy the build
+   directory between stages.
+
+ * In each stage directory loop through each subdirectory and then run each of the
+   install scripts it contains, again in alphanumeric order. These need to be named
+   with a two digit padded number at the beginning.
+   There are a number of different files and directories which can be used to
+   control different parts of the build process:
+
+     - **00-run.sh** - A unix shell script. Needs to be made executable for it to run
+
+     - **00-run-chroot.sh** - A unix shell script which will be run in the chroot
+       of the image build directory. Needs to be made executable for it to run.
+
+     - **00-debconf** - Contents of this file are passed to debconf-set-selections
+       to configure things like locale, etc.
+
+     - **00-packages** - A list of packages to install. Can have more than one, space
+       separated, per line.
+
+     - **00-packages-nr** - As 00-packages, except these will be installed using
+       the ```--no-install-recommends -y``` parameters to apt-get
+
+     - **00-patches** - A directory containing patch files to be applied
+
```

Comment:
Do you know what order the scripts are processed in? If there's both a 00-run.sh and 00-run-chroot.sh, which gets executed first? This question applies to the all the directories. Knowing the order of evaluation may be useful to some.

Reply:
The 00 is just an example, they should really all be numbered uniquely

Reply:
I guess it would depend on if it's doing a sort of all entries, or looping through each type and sorting those. If it's the former, then indicate that no number should be reused in any name in that directory. If it's processing xx-patches before xx-run-chroot.sh, indicate that patches are done first, then xx-run.sh, then xx-run-chroot.sh, etc.

Your additions are good stuff, so please don't think I'm bashing on them (although we do use bash to run them :))